salt/utils/event.py

```diff
@@ -259,7 +260,7 @@ def __load_uri(self, sock_dir, node):
         )
         return puburi, pulluri
 
-    def subscribe(self, tag, match_type=None):
+    def subscribe(self, tag=None, match_type=None):
```

Comment:
Before my PR, `subscribe` was a No-Op. It did nothing because `__init__` already connected it and the only code in here was to connect. Further back in the history of the code I can see it used to `ZMQ_SUBSCRIBE` to specific tags but it was removed such that ALL tags were listened to by default. That's when this function started to become a No-Op.

Behaviour wise, before my PR, `get_event` would not cache any entries at all
With this change in this PR, it would now cache ALL events.
To have true previous behaviour, without a tag we should do nothing - a warning printed maybe

To give a little background:
The goal was eventually that every call to subscribe has a call to unsubscribe to prevent memory leaks but I wanted to keep initial concept and implementation straight forward and allow it to be built upon.
I understood the subscribe and unsubscribe were meant to be doing this - telling the library we want those specific types of tags monitored, but were just broken since the library was monitoring everything, and the library further broken as it did not account for multiple tags in `get_event` in the way it discarded them.
Ideally in the future I see subscribe/unsubscribe as reference counting, so multiple calls to subscribe and single call to unsubscribe with the same parameters doesn't break (at moment it does), and also that subscribe monitors the callers name, and if we end script without unsubscribing we somehow dump the subscribers to detect leakers (it's how I've done it before in other languages - not sure on exact python way).
My PR was just to get the batching working though which needed the events caching
Hope I've managed to relay accurately my knowledge here.